accessible/fieldview.component.js

```diff
@@ -0,0 +1,146 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Angular2 Component that details how a Blockly.Field is
+ * rendered in the toolbox in AccessibleBlockly. Also handles any interactions
+ * with the field.
+ * @author madeeha@google.com (Madeeha Ghori)
+ */
+blocklyApp.FieldView = ng.core
+  .Component({
+    selector: 'field-view',
+    template: `
+    <li [id]="idMap['listItem']" role="treeitem" *ngIf="isTextInput(field)" 
+        [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr('blockly-argument-input', idMap['input'])" 
+        [attr.aria-level]="level"  aria-selected=false>
+      <input [id]="idMap['input']" [ngModel]="field.getValue()" (ngModelChange)="field.setValue($event)">
+    </li>
+    <li [id]="idMap['listItem']" *ngIf="isDropdown(field)"
+        [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr('blockly-argument-menu', idMap['label'])" 
+        [attr.aria-level]="level" aria-selected=false role="treeitem">
+      <label [id]="idMap['label']">{{stringMap['CURRENT_ARGUMENT_VALUE']}} {{field.getText()}}</label>
+      <ol role="group" [attr.aria-level]="level+1">
+        <li [id]="idMap[optionValue]" role="treeitem" *ngFor="#optionValue of getOptions(field)" 
+            [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr(idMap[optionValue + 'Button'], 'blockly-button')" 
+            [attr.aria-level]="level+1" aria-selected=false>
+          <button [id]="idMap[optionValue + 'Button']" (click)="handleDropdownChange(field,optionValue)">
+            {{optionText[optionValue]}}
+          </button>
+        </li>
+      </ol>
+    </li>
+    <li [id]="idMap['listItem']" role="treeitem" 
+        *ngIf="isCheckbox(field)" [attr.aria-level]="level" 
+        aria-selected=false>
+      // Checkboxes not currently supported.
+    </li>
+    <li [id]="idMap['listItem']" role="treeitem" 
+        [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr('blockly-argument-text', idMap['label'])" 
+        *ngIf="isTextField(field) && hasVisibleText(field)" 
+        [attr.aria-level]="level" aria-selected=false>
+      <label [id]="idMap['label']">
+        {{field.getText()}}
+      </label>
+    </li>
+    `,
+    inputs: ['field', 'level', 'index', 'parentId'],
+    providers: [blocklyApp.TreeService, blocklyApp.UtilsService],
+  })
+  .Class({
+    constructor: [blocklyApp.TreeService, blocklyApp.UtilsService, 
+        function(_treeService, _utilsService) {
+      this.optionText = {
+        keys: []
+      };
+      this.text = '';
```

Comment:
I seem to remember commenting on this before. Where is this used? If it is not used, can this line be removed?